src/main/java/online/book/store/service/shoppingcart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package online.book.store.service.shoppingcart;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import online.book.store.dto.request.cartitem.CartItemRequestDto;
+import online.book.store.dto.request.cartitem.UpdateRequestCartItemDto;
+import online.book.store.dto.response.shoppingcart.ShoppingCartResponseDto;
+import online.book.store.exception.EntityNotFoundException;
+import online.book.store.mapper.CartItemMapper;
+import online.book.store.mapper.ShoppingCartMapper;
+import online.book.store.model.Book;
+import online.book.store.model.CartItem;
+import online.book.store.model.ShoppingCart;
+import online.book.store.model.User;
+import online.book.store.repository.book.BookRepository;
+import online.book.store.repository.cartitem.CartItemRepository;
+import online.book.store.repository.shoppingcart.ShoppingCartRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartResponseDto getShoppingCart(String email) {
+        return shoppingCartMapper.toDto(shoppingCartRepository.findByUserEmail(email));
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartResponseDto addItem(Long id, CartItemRequestDto requestDto) {
+        Book book = bookRepository.findById(requestDto.getBookId())
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "Can't get book with id: " + requestDto.getBookId()));
+
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(id)
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "Can't get shopping cart with id: " + id));
+
+        CartItem cartItem = cartItemMapper.toEntity(requestDto);
+        cartItem.setBook(book);
+        cartItem.setShoppingCart(shoppingCart);
+
+        CartItem savedItem = cartItemRepository.save(cartItem);
+        shoppingCart.getCartItems().add(savedItem);
+
+        return shoppingCartMapper.toDto(shoppingCart);
```

Comment:
I'm not sure, but in this case we will return response like it 
{
    "id": 1,
    "userId": 3,
    "cartItems": null
}

maybe it would be better return cartItemDto 
{
    "id": 6,
    "bookId": 1,
    "bookTitle": "To Kill a Mockingbird",
    "quantity": 5
}
but it's only my opinion, I'm not sure

Reply:
In this case I return a shopping cart with a list of cart items, in case null, I can't receive as in a request you can't add null values
<img width="1677" alt="Screenshot 2024-05-19 at 13 59 58" src="https://github.com/vikkushnir/Book-store/assets/158613560/7997852f-2558-4e3e-a28a-a2df79ac4779">